frontend/src/components/details/DetailsComponent.js

```diff
@@ -1,24 +1,54 @@
 /**
- * the main component for detail page which includes  
- * canvas page and variable drop down list  
- * 
- * @author parham, 08.05  
+ * the main component for detail page which includes
+ * canvas page and variable drop down list
+ *
+ * @author parham, 08.05
  */
 
-import React, {Component, useContext} from 'react';
+import React, { Component, useContext } from 'react';
 import Canvas from './CanvasComponent';
 import SelectVariable from './SelectVariableComponent';
-import {PrivateSectionContext} from "hooks/PrivateSectionContext";
+import { PrivateSectionContext } from 'hooks/PrivateSectionContext';
+import DividerPannel from './PanelComponent';
+
+class DetailComponent extends Component {
+    state = {
+        compareCanvas: false
+    };
+    render() {
+        /*
+        we should later on find a way to make this line also compatible with class component 
+        */
+        //const [selectedProducts, setSelectedProducts] = useContext(PrivateSectionContext);
+
+        /*
+         the default canvas has to be divided into two canvases
+         an extra drom down button for second variable should be rendered 
+         the compare button should be disabled 
+         */
+        let handleCompareButton = () => {
+            const compareCanvas = true;
+            /*
+            now all components such as 
+            canvas component should be notified 
+            by setting the compareCanvas state to true
+            */
+            this.setState({ compareCanvas });
+        };
```

Comment:
you could do this also by using a context Provider in the function, then you wouldn't have to declare all these functions. 🤔 

If you use "should" in a documentation comment somewhere, since you mean it is not yet there, you could add a "**TODO:**" beforehand  so that we can easily find all areas that "should" still be worked on further somewhere in the near future😉

Reply:
you are right about the "TODO:". thanks for the hint.

Reply:
by "all functions", you mean "setState()" ?

Reply:
yes! exactly :)